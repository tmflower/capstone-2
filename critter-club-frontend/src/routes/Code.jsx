import { useState, useEffect } from "react";
import usersAPI from "../api/usersAPI";
import { useParams } from "react-router-dom";

export function Code() {
    
    const params = useParams();
    const parentUsername = params.username;

    const [accessCode, setAccessCode] = useState(1111);

    useEffect(() => {
        async function getCode() {    
            setAccessCode(await usersAPI.getCode(parentUsername));
        }

        getCode();
    },[parentUsername]);

    return (
        <div>
        {accessCode ? <p>Your access code is: {accessCode}</p> : null}
        <p>Your child will need to enter this code in order to set up a user account.</p>
        </div>
    )
}