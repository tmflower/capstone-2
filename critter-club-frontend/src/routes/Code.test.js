import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Code } from './Code';

// smoke test
test("renders without crashing", () => {
    render(
    <MemoryRouter>
      <Code/>
    </MemoryRouter>);
  });

// snapshot test
test("matches snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Code />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});