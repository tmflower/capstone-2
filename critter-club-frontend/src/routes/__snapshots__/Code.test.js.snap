// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 css-180j490-MuiPaper-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Your child will need to enter this code in order to set up a user account.
    </p>
    <a
      class="link"
      exact="true"
      href="/signup"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Child signup
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </a>
  </div>
</DocumentFragment>
`;
